cdap-ui/app/cdap/components/DataPrep/TopPanel/index.js

```diff
@@ -88,6 +94,56 @@ export default class DataPrepTopPanel extends Component {
     );
   }
 
+  onSubmit() {
+    if (this.props.onSubmit) {
+      let directives = DataPrepStore.getState().dataprep.directives;
+      let workspaceId = DataPrepStore.getState().dataprep.workspaceId;
```

Comment:
There is some error logic that might not be generic. As of now only the api calls are same. Can we see how we can move it to helper in the next iteration?